### PR TITLE
Support Wagtail 4.2 through 5.1

### PR DESCRIPTION
**Pull request by willbarton**
This change adds support for Wagtail 4.2 through 5.1. Wagtail 5.2 changes the breadcrumbs implementation in the admin, and support for it will follow separately.

This drops support for Wagtail 4.1.

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [x] Passes all existing automated tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ Wagtail-TreeModelAdmin is an extension for Wagtail's [ModelAdmin](http://docs.wa
 
 - Python 3.8+
 - Django 3.2 (LTS), 4.1 (current)
-- Wagtail 4.0+
+- Wagtail 5+,<5.2
```
Comment (chosak) [inline README.md:25]: Shouldn't this include 4.2 if we're maintaining support for now?

File: tox.ini [hunk 2]
```diff
@@ -44,11 +46,11 @@ commands=
     coverage xml
     diff-cover coverage.xml --compare-branch=origin/main --fail-under=100
 
-[testenv:interactive]
+[testenv:interactive
```
Comment (chosak) [inline tox.ini]: Is this a mistake?

```suggestion
[testenv:interactive]
```

File: treemodeladmin/tests/settings.py [hunk 3]
```diff
@@ -35,33 +33,15 @@
     "wagtail.sites",
     "wagtail.users",
     "wagtail.contrib.styleguide",
+    "wagtail",
+    "wagtail.test.testapp",
```
Comment (chosak) [inline treemodeladmin/tests/settings.py:37]: Does this project actually need testapp for any reason? I don't see any references to it in the tests anywhere. I think removing this also lets us remove the `WAGTAILADMIN_RICH_TEXT_EDITORS` definition below.

Including this also generates an error for me when I run `tox -e interactive`:

> django.db.migrations.exceptions.InconsistentMigrationHistory: Migration tests.0011_modelwithnullableparentalkey is applied before its dependency tests.0010_alter_customimage_file_and_more on database 'default'.

**Review by chosak — COMMENTED** [review]
I noticed something odd when playing with the test data in `tox -e interactive`. If you delete the only book under "Roald Dahl", you get sent to the URL http://localhost:8000/admin/treemodeladmintest/book/?author=2 which shows a strange and unexpected message:

image

It's unclear what the "0 out of 3" refers to, and the search message is strange. I don't think this behavior was introduced with these changes, but wanted to flag in case it was something new you hadn't seen.

**Comment by willbarton** [comment]
> It's unclear what the "0 out of 3" refers to, and the search message is strange. I don't think this behavior was introduced with these changes, but wanted to flag in case it was something new you hadn't seen.

Ok, so. I think this is from using ModelAdmin, and filtering the query it uses for the "Book" model (in this case). 

So, the "Author" page is actually a ModelAdmin for "Book" that sees there are 3 total books left, but 0 are displayed.

This isn't new behavior here, but it's definitely interesting! I'll have to dig a little deeper into this issue. But that might be best saved for when transitioning to wagtail-modeladmin for 5.2.

File: treemodeladmin/tests/settings.py (hunk 3 above)
Comment (willbarton) [reply treemodeladmin/tests/settings.py]: Removed in 5f84cd0

File: README.md (hunk 1 above)
Comment (willbarton) [reply README.md]: Fixed in 5f84cd0

**Review by chosak — APPROVED** [review]

**Pull request merged**